core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java

```diff
@@ -367,6 +367,11 @@ public CompletableFuture<Message> query(Message request) {
   @Override
   public void close() {
     mClosed = true;
+    MetricsSystem.removeMetrics(MetricKey.MASTER_EMBEDDED_JOURNAL_SNAPSHOT_LAST_INDEX.getName());
```

Comment:
can we make it a separate method?

Reply:
Looks like methods in `MetricsUtils` are general utilities of metrics.
I'm not sure if `MetricsUtils.clearJournalStateMachineMetrics()` is general enough.
The `MetricKey` got cleared is related to what's registered in `JournalStateMachine`.
How about add a private method `JournalStateMachine.clearMetrics()`?